### ISSUE TITLE
Statically allocating space for 16 headers is too few for redirects from some OAuth providers

### ISSUE THREAD

**Issue by mrodz** [FabianLars/tauri-plugin-oauth#21]
https://github.com/FabianLars/tauri-plugin-oauth/blob/50dadbf4a81cba51f625587b7722892f0b4316a6/src/lib.rs#L114

Google's OAuth API for web apps will send more than 16 headers, which causes this line:

https://github.com/FabianLars/tauri-plugin-oauth/blob/50dadbf4a81cba51f625587b7722892f0b4316a6/src/lib.rs#L116

To throw:
```
thread '<unnamed>' panicked at desktop\src\auth.rs:118:28:
called `Result::unwrap()` on an `Err` value: TooManyHeaders
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

For my use case.

Is this behavior present and noticeable for other users of this plugin?

**Comment by mrodz** [comment]
**A quick note:**
I did not see any stack trace when originally using this plugin directly and was only met with an OAuth redirect to a: `127.0.0.1:PORT did not send any data. net::ERR_EMPTY_RESPONSE` chromium page. I only found out this was the source of the bug by vendoring the dependency and manually doing some testing.

**Comment by FabianLars** [comment]
> Is this behavior present and noticeable for other users of this plugin?

For what it's worth, you're the first one to tell me about this. What's weird is that google is exactly the reason i created this plugin so it's probably the most tested oauth provider - i guess something changed.

Do you have a rough estimate on the amount of headers?

**Comment by mrodz** [comment]
On a fresh Chrome profile, the redirect sends exactly 16 headers. However, some users may use browser extensions or have an application-level firewall that add extra headers to HTTP requests, which is how I ended up with some in the 17-19 range and a TCP server crash without warning or message. I added space for 32 headers in my own app just to account for the case where there are extra headers.

Whether or not this is important enough to warrant additional space allocated in the handler function is up to you. I just wanted to bring this to your attention :)
